### PR TITLE
improve error handling on NOT upload a file/select template

### PR DESCRIPTION
**Pull request by rrchai**
fix issue #29: if user not upload file and press validation button, disconnect from server
fix issue #119: if user not select template and press validation button, disconnect from server

### CODE REVIEW

File: app.R [hunk 1]
```diff
@@ -528,16 +525,32 @@ schema_to_display_lookup <- data.frame(schema_name, display_name)
...
         actionButton("submitButton", "Submit to Synapse")
       })
 
     }
+    }
+    
+    ### format output text
+    output$text2 <- renderUI({
+      # test if template is selected and filled manifest is uploaded
+      shiny::validate(need(!is.null(input$template_type), "Please choose a template in Select your Dataset tab !!!"),
```
Comment (sujaypatil96) [inline app.R:541]: Change font color of message to red.

Error message: "Please select a template from the 'select your dataset tab'".

File: app.R [hunk 2]
```diff
@@ -528,16 +525,32 @@ schema_to_display_lookup <- data.frame(schema_name, display_name)
...
+                      need(!is.null(rawData()), "Please upload a filled metadata !!!")
```
Comment (sujaypatil96) [inline app.R:542]: Error message: "Please upload a filled metadata template".

File: app.R [hunk 3]
```diff
@@ -528,16 +525,32 @@ schema_to_display_lookup <- data.frame(schema_name, display_name)
       show('gsheet_btn')
       
     } else {
-      output$text2 <- renderUI({
-        HTML("Your metadata is valid!")
-      })
-
+      
+      validation_res <- "Valid"
```
Comment (sujaypatil96) [inline app.R:529]: "valid" (not titlecase).

**Review by sujaypatil96 — CHANGES_REQUESTED** [review]

File: app.R (hunk 3 above)
Comment (rrchai) [reply app.R]: update in b1d9197

File: app.R (hunk 1 above)
Comment (rrchai) [reply app.R]: update message in b1d9197
change color to red in 4d26a50

File: app.R (hunk 2 above)
Comment (rrchai) [reply app.R]: update message in b1d9197

**Pull request merged**